controllers/dockercluster_controller.go

```diff
@@ -38,72 +48,113 @@ type DockerClusterReconciler struct {
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=dockerclusters,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=dockerclusters/status,verbs=get;update;patch
 
-// Reconcile handles DockerCluster events
-func (r *DockerClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, rerr error) {
+// Reconcile reads that state of the cluster for a DockerCluster object and makes changes based on the state read
+// and what is in the DockerCluster.Spec
+func (r *DockerClusterReconciler) Reconcile(request reconcile.Request) (_ reconcile.Result, reterr error) {
 	ctx := context.Background()
-	log := r.Log.WithValues("dockercluster", req.NamespacedName)
-	log.Info("Reconciling cluster")
+	logger := log.Log.WithName(controllerName).
+		WithName(fmt.Sprintf("namespace=%s", request.Namespace)).
+		WithName(fmt.Sprintf("dockerCluster=%s", request.Name))
 
+	// Fetch the DockerCluster instance
 	dockerCluster := &infrastructurev1alpha2.DockerCluster{}
-	if err := r.Client.Get(ctx, req.NamespacedName, dockerCluster); err != nil {
+	if err := r.Client.Get(ctx, request.NamespacedName, dockerCluster); err != nil {
 		if apierrors.IsNotFound(err) {
 			return ctrl.Result{}, nil
 		}
-		log.Error(err, "failed to get DockerCluster")
```

Comment:
Is this redundant? Will controller-runtime print a traceable error here or should we still log this?

Reply:
this is redundant (unless we log an error with a stack trace)